controllers/remote/cluster_cache_tracker.go

```diff
@@ -155,22 +195,47 @@ func (t *ClusterCacheTracker) getClusterAccessorLH(ctx context.Context, cluster
 
 // newClusterAccessor creates a new clusterAccessor.
 func (t *ClusterCacheTracker) newClusterAccessor(ctx context.Context, cluster client.ObjectKey, indexes ...Index) (*clusterAccessor, error) {
+	log := ctrl.LoggerFrom(ctx)
+
 	// Get a rest config for the remote cluster
 	config, err := RESTConfig(ctx, clusterCacheControllerName, t.client, cluster)
 	if err != nil {
 		return nil, errors.Wrapf(err, "error fetching REST client config for remote cluster %q", cluster.String())
 	}
 
-	// Create a mapper for it
-	mapper, err := apiutil.NewDynamicRESTMapper(config)
+	// Create a client and a mapper for the cluster.
+	c, mapper, err := t.createClient(config, cluster)
 	if err != nil {
-		return nil, errors.Wrapf(err, "error creating dynamic rest mapper for remote cluster %q", cluster.String())
+		return nil, err
 	}
 
-	// Create the client for the remote cluster
-	c, err := client.New(config, client.Options{Scheme: t.scheme, Mapper: mapper})
+	// Detect if the controller is running on the cluster.
+	runningOnCluster, err := t.runningOnCluster(ctx, c, cluster)
```

Comment:
runningOnWorkloadCluster?

Reply:
@ykakarap are you able to address this so we can get this pulled in? Stefan is OOF afaik

Reply:
`runningOnCluster` seems to be consistent with the naming used in other parts of this file `ClusterCacheTracker`, `clusterAccessor`, etc. Looks like cluster here always refers to workload cluster.

Reply:
I am happy to rename it if other have strong opinions :)

Reply:
+1 for not renaming if it's consistent and assumed that clusters == workload cluster in this file

Reply:
+1 for renaming
for consistency, let's fix all comments replacing "running on a cluster" with "running on a workload cluster"

Reply:
func has been renamed